pilot/pkg/proxy/envoy/v2/externalservice.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	networking "istio.io/api/networking/v1alpha3"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/proxy/envoy/v1"
+)
+
+func buildExternalServiceTCPListeners(mesh *meshconfig.MeshConfig, config model.IstioConfigStore) ([]*xdsapi.Listener, v1.Clusters) {
```

Comment:
In a separate iteration, I suggest a bit of cleanup - no need to return clusters for example.

Reply:
Yes, absolutely. This is just a minimum viable product, there are many places it can be trimmed and improved. I wanted to keep the changes minimal to focus on correctness initially.